lib/jekyll-crds/filters/meta_filters.rb

```diff
@@ -44,6 +44,12 @@ def meta_image(page)
       "https:#{escape(image_url)}"
     end
 
+    def get_canonical_host(page)
+      host = page['distribution_channels'].detect { |channel| channel['canonical'].present? }
```

Comment:
Could you add a check just in case `page['disctribution_channels']` ins `nil`?

- You could return early (that's my style)
- Set `page['disctribution_channels']` to `[]` (or whatever it's empty state should be)

Reply:
This logic that will be in place in every repo will handle this, unless I am not understanding exactly what you are asking?
```
 {% if page.distribution_channels %}
    <link rel="canonical" href="{{ page | get_canonical_host }}">
  {% endif %}
```
Also with JBs distribution channel if there is not one selected he fixed the logic to make crds.net the default

Reply:
@silasbaur - You are correct that as long as you do the if check-in Jekyll you should be fine. I'm more thinking if the check is forgotten or if it's used in a way you are not expecting in the future.

More of a defensive measure I think in this situation is helpful. If you are building something that is distributed among multiple places or handling input you may not always have control of.

I hit the error while testing without the if statement and it was somewhat hard to debug it.

I had added something like this `return '' if page['distribution_channels'].nil?` to make it work.

If you want to talk through it let me know.